src/Mod/Sketcher/App/planegcs/Constraints.cpp

```diff
@@ -2542,38 +2285,18 @@ void ConstraintPointOnParabola::errorgrad(double* err, double* grad, double* par
     }
 }
 
-double ConstraintPointOnParabola::error()
-{
-    double err;
-    errorgrad(&err, nullptr, nullptr);
-    return scale * err;
-}
-
-double ConstraintPointOnParabola::grad(double* param)
-{
-    // first of all, check that we need to compute anything.
-    if (findParamInPvec(param) == -1) {
-        return 0.0;
-    }
-
-    double deriv;
-    errorgrad(nullptr, &deriv, param);
-
-    return deriv * scale;
-}
-
 
 // --------------------------------------------------------
 // ConstraintAngleViaPoint
 ConstraintAngleViaPoint::ConstraintAngleViaPoint(Curve& acrv1, Curve& acrv2, Point p, double* angle)
+    : crv1(acrv1.Copy())
+    , crv2(acrv2.Copy())
 {
     pvec.push_back(angle);
     pvec.push_back(p.x);
     pvec.push_back(p.y);
     acrv1.PushOwnParams(pvec);
     acrv2.PushOwnParams(pvec);
```

Comment:
With `acrv1` and `acrv2` already copied into `crv1` and `crv2`, is this still correct?

Reply:
It works because `PushOwnParams` use pointer members of `acrv2` which are still the same after copy.
But I agree it needs to use `crv1` to clarify.